### PR TITLE
Fix MS15574: Tiny memory leak in Wallet.h

### PR DESCRIPTION
**Pull request by zfox23**
Fixes [MS15574](https://highfidelity.manuscript.com/f/cases/15574/Tiny-memory-leak-in-Wallet-h).

### CODE REVIEW

**Comment by zfox23** [comment]
## Test Plan
1. Using this build, authenticate your Wallet. Close and re-open the Wallet app, verifying that you're still authenticated.
2. Make a Purchase from the Marketplace (it can be a free item). Verify the purchase goes through OK.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10344/HighFidelity-Beta-PR13295-7407507.apk

File: interface/src/commerce/Wallet.cpp [hunk 1]
```diff
@@ -359,6 +360,7 @@ void Wallet::clear() {
     // tell the provider we got nothing
     updateImageProvider();
     _passphrase->clear();
+    delete _passphrase;
```
Comment (AndrewMeadows) [inline interface/src/commerce/Wallet.cpp]: I see other code that checks the value of `_passphrase` before dereferencing it (for example: `Wallet::setPassphrase()`), however the changes above don't set `_passphrase` to `nullptr` after deleting that to which it points.  This looks like a crash just waiting to happen.

`_passphrase` should be set to `nullptr` here.

Comment (zfox23) [reply interface/src/commerce/Wallet.cpp]: now that you make this comment, i'm realizing that this isn't even the right place to put this code. @AndrewMeadows: check out the new commit and let me know if this is still wrong, please.

Comment (davidkelly) [reply interface/src/commerce/Wallet.cpp]: nice catch @AndrewMeadows

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10346/HighFidelity-Beta-PR13295-60aff7a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10346/HighFidelity-Beta-PR13295-60aff7a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10346/HighFidelity-Beta-Interface-PR13295-60aff7a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10346/HighFidelity-Beta-PR13295-60aff7a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10346/HighFidelity-Beta-Interface-PR13295-60aff7a.exe

**Comment by QliCknox** [comment]
Currently testing

**Comment by QliCknox** [comment]
Pass @conklin94122 

Wallets can be authenticated and remain authenticated, even after the wallet has been closed multiple times. User can buy items from the marketplace as expected.

**Pull request merged**